File: STACK/LinkedStack.java
package STACK;

class Stack {
    public int data;
    public Stack pre;
    public Stack top;

    public Stack() {
        data = 0;
        pre = null;
        top = null;
    }
    public Stack(int data) {
        this.data = data;
    }

    public boolean isEmpty() {
        if(top == null) {
            return true;
        }
        return false;
    }
    public void push(int data) {
        Stack newStack = new Stack(data);
        newStack.pre = top;
        top = newStack;
    }
    public int pop() {
        int data;
        Stack newStack = top;
        if(isEmpty()) {
            System.out.println("Stack is Empty");
            return 0;
        }
        data = newStack.data;
        top = newStack.pre;
        return data;
    }
}

class LinkedStack {
    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(10);
        stack.push(20);
        stack.push(30);
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
    }
}

File: STACK/LinkedStack.kt
class Stack {
    var data: Int
    var pre: Stack? = null
    var top: Stack? = null

    constructor() {
        data = 0
        pre = null
        top = null
    }

    constructor(data: Int) {
        this.data = data
    }

    val isEmpty: Boolean
        get() {
            if (top == null) {
                return true
            }
            return false
        }

    fun push(data: Int) {
        val newStack = Stack(data)
        newStack.pre = top
        top = newStack
    }

    fun pop(): Int {
        val data: Int
        val newStack = top
        if (isEmpty) {
            println("Stack is Empty")
            return 0
        }
        data = newStack!!.data
        top = newStack.pre
        return data
    }
}

internal object LinkedStack {
    @JvmStatic
    fun main(args: Array<String>) {
        val stack = Stack()
        stack.push(10)
        stack.push(20)
        stack.push(30)
        println(stack.pop())
        println(stack.pop())
        println(stack.pop())
        println(stack.pop())
    }
}